pybaseball/statcast_pitcher.py

```diff
@@ -25,3 +29,78 @@ def statcast_pitcher(start_dt: Optional[str] = None, end_dt: Optional[str] = Non
+    res = requests.get(url, timeout=None).content
+    data = pd.read_csv(io.StringIO(res.decode('utf-8')))
+    return data
+
+@cache.df_cache()
+def statcast_pitcher_pitch_arsenal(year: int, minP: int = 250, arsenal_type: str = "avg_speed") -> pd.DataFrame:
+    arsenals = ["avg_speed", "n_", "avg_spin"]
+    if arsenal_type not in arsenals:
+        print(f"Not a valid arsenal_type. Must be one of {', '.join(arsenals)}.")
+        return
+    url = f"https://baseballsavant.mlb.com/leaderboard/pitch-arsenals?year={year}&min={minP}&type={arsenal_type}&hand=&csv=true"
+    res = requests.get(url, timeout=None).content
+    data = pd.read_csv(io.StringIO(res.decode('utf-8')))
+    return data
+
+@cache.df_cache()
+def statcast_pitcher_arsenal_stats(year: int, minPA: int = 25) -> pd.DataFrame:
+    # test to see if pitch types needs to be implemented or if user can subset on their own
+    url = f"https://baseballsavant.mlb.com/leaderboard/pitch-arsenal-stats?type=pitcher&pitchType=&year={year}&team=&min={minPA}&csv=true"
+    res = requests.get(url, timeout=None).content
+    data = pd.read_csv(io.StringIO(res.decode('utf-8')))
+    return data
+
+@cache.df_cache()
+def statcast_pitcher_pitch_movement(year: int, minP: Union[int, str] = "q", pitch_type: str = "FF") -> pd.DataFrame:
+    pitch_types = ["FF", "SIFT", "CH", "CUKC", "FC", "SL", "FS", "ALL"]
+    if pitch_type not in pitch_types:
+        print(f"Not a valid pitch_type. Must be one of {', '.join(pitch_types)}")
+        return
+    url = f"https://baseballsavant.mlb.com/leaderboard/pitch-movement?year={year}&team=&min={minP}&pitch_type={pitch_type}&hand=&x=pitcher_break_x_hidden&z=pitcher_break_z_hidden&csv=true"
+    res = requests.get(url, timeout=None).content
+    data = pd.read_csv(io.StringIO(res.decode('utf-8')))
+    return data
+
+@cache.df_cache()
+def statcast_pitcher_active_spin(year: int, minP: int = 250) -> pd.DataFrame:
+    url = f"https://baseballsavant.mlb.com/leaderboard/active-spin?year={year}&min={minP}&hand=&csv=true"
+    res = requests.get(url, timeout=None).content
+    data = pd.read_csv(io.StringIO(res.decode('utf-8')))
+    return data
+
+@cache.df_cache()
+def statcast_pitcher_percentile_ranks(year: int) -> pd.DataFrame:
+    url = f"https://baseballsavant.mlb.com/leaderboard/percentile-rankings?type=pitcher&year={year}&position=&team=&csv=true"
+    res = requests.get(url, timeout=None).content
+    data = pd.read_csv(io.StringIO(res.decode('utf-8')))
+    # URL returns a null player with player id 999999, which we want to drop
+    return data.loc[data.player_name.notna()].reset_index(drop=True)
+
+@cache.df_cache()
+def statcast_pitcher_spin_dir_comp(year: int, pitch_combo: str = "4 - Seamer / Changeup", minP: int = 100, pov: str = "Pit") -> pd.DataFrame:
+    pitches = ["4-Seamer", "Changeup", "Curveball", "Slider", "Sinker", "Cutter"]
```

Comment:
Fun fact: I checked the URLs a bit and this actually supports more pitches than this -- I think it supports the full range of the Statcast pitches. Splitter doesn't appear in the frontend, but it works.

Reply:
The one exception is 'All', it doesn't support that code

Reply:
In this case, is all just simulated with something like `','.join(pitches)` ?

Reply:
I think because it's comparing two different pitches it wouldn't support "All"